docs/cudf/source/developer_guide/benchmarking.md

```diff
@@ -0,0 +1,323 @@
+they should also assume as little as possible about the nature of the data.
+For instance, unless there are meaningful functional differences,
+benchmarks should not care about the dtype or nullability of the data.
+Objects that differ in these ways should be interchangeable for most benchmarks.
+The goal of writing benchmarks in this way is to then automatically benchmark objects with different properties.
+We support this use case with the `benchmark_with_object` decorator.
+
+The use of this decorator is best demonstrated by example:
+
+```python
+@benchmark_with_object(cls="dataframe", dtype="int", cols=6)
+def bench_foo(benchmark, dataframe):
+    benchmark(dataframe.foo)
+```
+
+In the example above `bench_foo` will be run for DataFrames containing six columns of integer data.
+The decorator allows automatically parametrizing the following object properties:
+
+- `cls`: Objects of a specific class, e.g. `DataFrame`.
+- `dtype`: Objects of a specific dtype.
+- `nulls`: Objects with and without null entries.
+- `cols`: Objects with a certain number of columns.
+- `rows`: Objects with a certain number of rows.
+
+In the example, since we did not specify the number of rows or nullability,
+it will be run once for each valid number of rows and for both nullable and non-nullable data.
+The valid set of all parameters (e.g. the numbers of rows) is stored in the `common/config.py` file.
+This decorator allows a developer to write a generic benchmark that works for many types of objects,
+then have that benchmark automatically run for all objects of interest.
+
+### Parametrizing tests
+
+The `benchmark_with_object` decorator covers most use cases and automatically guarantees a baseline of benchmark coverage.
+However, many benchmarks will require more customized objects.
+In some cases those will be the primary targets whose methods are called.
+For instance, a benchmark may require a `Series` with a specific data distribution.
+In others, those objects will be arguments passed to other functions.
+An example of this is `DataFrame.where`, which accepts many types of objects to filter with.
+
+In the first case, fixtures should follow certain rules.
+When writing fixtures, developers should make the data sizes dependent on the benchmarks configuration.
+The `benchmarks/common/config.py` file defines standard data sizes to be used in benchmarks.
+These data sizes can be tweaked for debugging purposes (see [](#testing-benchmarks) below).
+Fixture sizes should be relative to the `NUM_ROWS` and/or `NUM_COLS` variables defined in the config module.
+These rules ensure consistency between these fixtures and those provided by `benchmark_with_object`.
+
+In the second case, fixtures are really functioning as parameters, which we discuss in the next section.
+
+
+### Parametrization: custom fixtures, `pytest-cases` and `pytest.mark.parametrize`
+
+When it comes to parametrizing tests, we have a number of options at our disposal.
```

Comment:
```suggestion
When it comes to parametrizing benchmarks, we have a number of options at our disposal.
```

?

Reply:
Ha yes sorry about that. Thanks for the fix.